src/functions/pattern.ts

```diff
@@ -1,5 +1,25 @@
 import { IFunction, IFunctionResult, IRulePatternOptions } from '../types';
 
+function test(value: string, regex: any) {
+  let re;
+  if (typeof regex === 'string') {
+    // regex in a string like {"match": "/[a-b]+/im"} or {"match": "[a-b]+"} in a json ruleset
+    // the available flags are "gimsuy" as described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+    const splitRegex = /^\/(.+)\/([gimsuy]*)$/.exec(regex);
```

Comment:
I wouldn't be less lenient about flags here.
```suggestion
    const splitRegex = /^\/(.+)\/([a-z]*)$/.exec(regex);
```
IMHO we don't have to be so defensive here. Furthermore, we won't need to adjust the code in the future when a new flag is introduced. (`s` is fairly new for instance, introduced in ES2018 IIRC)

Moreover, we don't seem to allow regular expressions without any flags specified, so I'd suggest tweaking the above regular expression further
```suggestion
    const splitRegex = /^\/(.+)\/([a-z]*)?$/.exec(regex);
```